### PR TITLE
Implement Calculator Framework for WHO-5 Score

### LINKED ISSUE #16: Implement Calculator Framework for Who-5 Score
1. Use Westley as reference/template for implementation. import { useCalculatorFramework, type CalculatorConfig, type CalculatorStep } from '@/composables/useCalculatorFramework';
2. Update unit tests and make sure tests passes
3. Create Playwright tests

### PR DESCRIPTION
**Pull request by iittcc**
## Summary
Migrates the WHO-5 Score component from traditional Vue pattern to use the Calculator Framework, following the PUQE component as a reference template.

### Changes Made
- **Component Migration**: Converted WHO-5Score.vue to use `useCalculatorFramework` composable
- **Framework Integration**: Replaced direct Vue refs with framework-based data binding patterns
- **Patient Data Handling**: Updated to use `framework.patientData` with proper defaults and validation
- **Form Submission**: Replaced manual form handling with `framework.submitCalculation()`
- **Test Migration**: Updated all 25 component tests to work with framework mocking patterns
- **Playwright Tests**: Created comprehensive end-to-end tests with helper classes and test data

### Technical Details
- Uses `useCalculatorFramework` composable with WHO-5 specific configuration
- Implements proper framework data binding with `framework.setFieldValue()`
- Maintains all existing functionality while adding framework benefits
- Follows established patterns from PUQE component implementation

### Test Coverage
- ✅ All 25 component tests passing with framework mocking
- ✅ Comprehensive Playwright tests for WHO-5 calculator
- ✅ Clinical accuracy tests for WHO-5 scoring algorithm
- ✅ Edge cases and boundary value testing
- ✅ Mobile responsiveness and accessibility tests

### Files Changed
- `src/components/Who-5Score.vue` - Component migrated to framework pattern
- `tests/components/Who-5Score.test.ts` - Tests updated for framework integration
- `tests/playwright/who5-score.spec.ts` - New Playwright tests
- `tests/playwright/helpers/who5-calculator-helper.ts` - New helper class
- `tests/playwright/fixtures/who5-test-data.ts` - Test data fixtures

## Test Plan
- [x] Unit tests pass (25/25)
- [x] Component tests pass  
- [x] Type checking passes
- [x] Linting passes
- [x] Framework integration verified
- [x] Playwright tests created (comprehensive coverage)
- [x] All existing functionality preserved
- [x] Patient data binding works correctly
- [x] Clinical calculations accurate

## Resolves
Closes #16

## Summary by CodeRabbit

* **Refactor**
  * Streamlined the WHO-5 Score Calculator component to use a centralized framework for managing state, data, and logic, resulting in a more consistent and maintainable user experience.

* **Tests**
  * Overhauled unit and integration tests to focus on framework integration and component behavior.
  * Added comprehensive end-to-end Playwright tests covering UI, scoring logic, clinical accuracy, accessibility, mobile responsiveness, performance, and error handling.
  * Introduced reusable test data fixtures and helper utilities for automated testing.

* **New Features**
  * Enhanced accessibility and mobile responsiveness for the WHO-5 Score Calculator.
  * Improved error handling and validation for user inputs.
  * Integrated toast notifications for user feedback within the WHO-5 Score Calculator interface.

### CODE REVIEW

File: src/components/Who-5Score.vue [hunk 1]
```diff
@@ -5,34 +5,102 @@
...
+              text: '5. I de sidste 2 uger har min daglig været fyldt med ting der interesserer mig',
+              optionsType: 'wellbeing'
+            }"
+            :options="wellbeingOptions"
+            :index="4"
+            :framework-answer="who5Data.question5"
+            :is-unanswered="formSubmitted && !who5Data.question5"
+            scrollHeight="18rem"
+            @update:answer="framework.setFieldValue('calculator', 'question5', $event)"
+          />
```
[the remark this answers is not in the document]
Comment (iittcc) [reply src/components/Who-5Score.vue]: resolved
